packages/anoncreds/src/services/registry/AnonCredsRegistry.ts

```diff
@@ -8,8 +8,14 @@ import type { GetRevocationRegistryDefinitionReturn } from './RevocationRegistry
 import type { GetSchemaReturn, RegisterSchemaOptions, RegisterSchemaReturn } from './SchemaOptions'
 import type { AgentContext } from '@aries-framework/core'
 
-// This service can be registered multiple times in a single AFJ instance.
+/**
+ * @public
+ */
 export interface AnonCredsRegistry {
+  // TODO: should this be a single regex? So just supportedMethodRegex. If you support multiple / has complex queries you can just combine them.
+  // This would prevent the need for a double loop (loop through all registries, then loop through all supportedMethods)
+  supportedMethods: RegExp[]
```

Comment:
I think here we should assume the most common use case, which I think is one identifier. Which regex it is rather easy to match on `A` or `B` (`(A|B)`).

Reply:
I think I'd prefer the `supportedIdentifier`, but I might prefer it if it indicates a regex. so `identifierPattern` or something? And we can provide a util function with would be something like this:

```typescript
const getSimpleMethodPattern(method:string): RegExp {
  return new RegExp(`^did:${method}:.*$`) // not a perfect pattern but you get the gist :).
}

Reply:
I like it!